src/NServiceBus.Core/Transports/Learning/DelayedMessagePoller.cs

```diff
@@ -50,9 +50,10 @@ public void Start()
                       {
                           await Task.Delay(TimeSpan.FromSeconds(1), polling.Token).ConfigureAwait(false);
                       }
-                      catch (OperationCanceledException)
+                      catch (Exception ex) when (ex.IsCausedBy(polling.Token))
                       {
-                          // polling is being stopped
+                          // private token, poller is being stopped, log the exception in case the stack trace is ever needed for debugging
+                          Logger.Debug("Operation canceled while stopping delayed message polling.", ex);
```

Comment:
What do you think about this? I remember we had a discussion a while back and concluded that "every stack trace will eventually be valuable".

Reply:
Actually, given the only operation here is the `Task.Delay`, it's perhaps a bit silly here. It's more relevant in the message pump which will have a deep-ish call stack underneath it.